share/wake/lib/core/list.wake

```diff
@@ -802,3 +835,30 @@ export def groupBy (cmp: a => a => Boolean) (list: List a): List (List a) =
             (elem, head, rest), tail
         _ = (elem, Nil), acc
     foldr combo Nil sorted
+
+export def optionToList: Option a => List a = match _
+    Some x = x, Nil
+    None = Nil
```

Comment:
I'd put this into `option.wake` instead, and while I did walk back a lot of its usage as discussed, we do still want an `optionToListFn` counterpart.

Reply:
I'll move that over to option.wake.

`optionToListFn` is not needed for this change so I'm not inclined to add it here. Let's discuss that in wake-dev maybe

Reply:
(or a separate PR that adds optionToList and optionToListFn, I can rebase optionToList in if that lands first)